src/monitor.py

```diff
@@ -105,9 +103,7 @@ def check_job_condition(self,
                             logs_on_error: bool = True,
                             ):
         try:
-            self.check_deployment_for_early_errors(job_resource_name(job.name, job.version))
-            check_until_job_is_operational(self._get_internal_job_url(job),
-                                           deployment_timestamp, on_job_alive)
+            check_until_job_is_operational(job, deployment_timestamp, on_job_alive)
```

Comment:
What about this whole logic behind `check_deployment_for_early_errors`? Don't we need to check the events and catch `Insufficient cpu` status and so on?

Reply:
I assumed these also appeared in `container_statuses`, but they appear in `container_conditions` before that pods have `container_statuses`.

Reply:
Your former version was also reading events and rollouts for some reason. So do we really get the same results now by not reading the k8s events? If so, that's brilliant, but it sounds too good to be true.

Reply:
The events were used to look for early errors. This was a catch-all place to look for early errors. If there was an early error before the liveness check, raise exception. 

Rollouts and events were also used to check if a deploy looked good, and if it did, control was passed to the liveness check.

Now the checking for early errors is done as part of the liveness check. We are looking for early errors in the same elements (reasons and messages), but once the liveness check looks good, we stop looking for early errors.

Should the same early error check be added to the ready check?

Reply:
> Should the same early error check be added to the ready check?

Ready endpoint is fine. 

I'm mostly concerned about the scenario where the cluster runs out of resources. As far as I remember there were no pods created in that case and the only root cause error could be found in the events. That's why with this new approach, without this early errors detection, if there's no pods, it goes straight to liveness probe, which surely won't respond and that will hang the whole `_wait_until_job_is_alive` for 15 minutes. That's something we wanted to avoid back then. 
I think calling both of the checks will do the trick:
```python
check_deployment_for_early_errors(job_resource_name(job.name, job.version))
check_until_job_is_operational(job, deployment_timestamp, on_job_alive)
```